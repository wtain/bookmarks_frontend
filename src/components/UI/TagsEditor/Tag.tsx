
import React, { useState } from "react";
import TagDto from "../../../domain/dto/TagDto";
import cl from './TagsEditor.module.css'

interface Props {
    tag: TagDto;
    onDelete?: () => void;
    onClick?: () => void;
}

const Tag: React.FC<Props> = (props: Props) => {

    const [showClose, setShowClose] = useState(false)
    const [clickablePointer, setClickablePointer] = useState(false);

    const clickable = cl.clickable;

    const styles = [cl.tag];

    if (clickablePointer) {
        styles.push(clickable);
    }

    return (
        <span className={styles.join(" ")} 
            onClick={() => {
                if (props.onClick) {
                    props.onClick()
                }
            }}
            onMouseEnter={() => {
                setShowClose(true);
                if (props.onClick) {
                    setClickablePointer(true);
                }
            }}
            onMouseLeave={() => {
                setShowClose(false);
                setClickablePointer(false);
            }}>
            {props.tag.name}
            {
                (showClose && props.onDelete) ? 
                    <div className={cl.close_button_wrapper}>
                        <button className={cl.close_button} onClick={() => {
                                if (props.onDelete !== null && props.onDelete !== undefined)
                                    props.onDelete!()
                            }}>
                            x
                        </button> 
                    </div>
                    : <></>
            }
        </span>
    )
}

export default Tag;