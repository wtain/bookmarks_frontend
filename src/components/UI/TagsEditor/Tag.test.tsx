import renderer from 'react-test-renderer';
import Tag from './Tag';

it('Tag renders correctly', () => {
  const tag = {
    name: "tag1",
    id: "some-id"
  };

  const tree = renderer
    .create(
      <Tag tag={tag} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});