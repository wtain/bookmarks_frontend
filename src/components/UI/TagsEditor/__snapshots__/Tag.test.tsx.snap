// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<span
  className="tag"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  tag1
</span>
`;
